#!/bin/bash
#PBS -S /bin/sh
#PBS -V
#PBS -l nodes=1:ppn=12,walltime=36:00:00
#PBS -N DG_dRNA_sngl_end
#PBS -l mem=24Gb
#PBS -o localhost:/scratch/$USER/$PBS_JOBID.$PBS_JOBNAME.jobout
#PBS -e localhost:/scratch/$USER/$PBS_JOBID.$PBS_JOBNAME.joberr
#PBS -M ra94@nyu.edu
#PBS -m abe
#PBS -q cgsb-s

module purge
module load bowtie/intel/1.0.1
module load bowtie2/intel/2.2.1
module load tophat/intel/2.0.11
module load samtools/intel/0.1.19

SOURCE=$source #this is the original fastq directory
DESTINATION=$destination # this is the destination directory
NAME=$name # this is the file name variable (everything before .fastq)
REF=$ref #this is the reference file prefix. It depends on the use of spike-ins. Options are "Scer3", "ScERCC", "ScBACERCC" or "ScBacERCCelegans"

cd /data/cgsb/gresham/LABSHARE/SHARED/CODE

#1. Do the optimized alignment and keep unaligned
bowtie2 -p 12 -N 1 -L 20 -i S,1.2,0.50 --local --score-min L,0,1.2  --un ${DESTINATION}/${NAME}_Bowtie2.fastq  -x /data/cgsb/gresham/LABSHARE/SHARED/REFERENCES/${REF}_bowtie2  -U ${SOURCE}/${NAME}.fastq  -S ${DESTINATION}/${NAME}_Bowtie2.sam


#2. Align the unmapped from the previous step with TopHat
tophat -I 3000 -p 12 --no-convert-bam -o ${DESTINATION}/${NAME}_TopHat /data/cgsb/gresham/LABSHARE/SHARED/REFERENCES/${REF}_bowtie2  ${DESTINATION}/${NAME}_Bowtie2.fastq

#3. Extract unmapped reads and save them
./ExtractUnmappedTophat.pl ${DESTINATION}/${NAME}_Bowtie2.fastq ${DESTINATION}/${NAME}_TopHat/accepted_hits.sam ${DESTINATION}/${NAME}_unaligned.fastq

#4. Prepare the two .sam files from steps 1 and 2 for merging and merge
java -Xmx4g -jar /share/apps/picard-tools/1.111/ReorderSam.jar INPUT=${DESTINATION}/${NAME}_Bowtie2.sam OUTPUT=${DESTINATION}/${NAME}_sortBowtie2.bam REFERENCE=/data/cgsb/gresham/LABSHARE/SHARED/REFERENCES/${REF}.fa

java -Xmx4g -jar /share/apps/picard-tools/1.111/ReorderSam.jar INPUT=${DESTINATION}/${NAME}_TopHat/accepted_hits.sam OUTPUT=${DESTINATION}/${NAME}_sortTopHat.bam REFERENCE=/data/cgsb/gresham/LABSHARE/SHARED/REFERENCES/${REF}.fa

samtools merge ${DESTINATION}/${NAME}.merged.bam ${DESTINATION}/${NAME}_sortBowtie2.bam ${DESTINATION}/${NAME}_sortTopHat.bam

#5. Filter for mapping quality
samtools sort ${DESTINATION}/${NAME}.merged.bam ${DESTINATION}/${NAME}.merged.sort

samtools view -bq 15 ${DESTINATION}/${NAME}.merged.sort.bam > ${DESTINATION}/${NAME}.MAPQ15.bam

#6. Remove duplicates

mkdir ${DESTINATION}/metrics
java -Xmx4g -jar /share/apps/picard-tools/1.111/MarkDuplicates.jar INPUT=${DESTINATION}/${NAME}.MAPQ15.bam OUTPUT=${DESTINATION}/${NAME}.noDUPS.bam ASSUME_SORTED=true METRICS_FILE=${DESTINATION}/metrics/${NAME}.duplication_metrics  REMOVE_DUPLICATES=true

samtools view -h ${DESTINATION}/${NAME}.MAPQ15.bam > ${DESTINATION}/${NAME}.MAPQ15.sam
samtools view -h ${DESTINATION}/${NAME}.noDUPS.bam > ${DESTINATION}/${NAME}.noDUPS.sam

#8. Make Metrics files
java -Xmx4g -jar /share/apps/picard-tools/1.111/CollectInsertSizeMetrics.jar HISTOGRAM_FILE=/data/cgsb/gresham/Niki/RESULTS/Growth/PRELIMINARY2/${TAG}.insert_histogram REFERENCE_SEQUENCE=/data/cgsb/gresham/Niki/RESULTS/A111sequencing/References/ScERCC.fa ASSUME_SORTED=true INPUT=/data/cgsb/gresham/Niki/RESULTS/Growth/PRELIMINARY2/${TAG}.MAPQ15.bam OUTPUT=/data/cgsb/gresham/Niki/RESULTS/Growth/PRELIMINARY2/${TAG}.insert_metrics

java -Xmx4g -jar /share/apps/picard-tools/1.111/CollectRnaSeqMetrics.jar INPUT=/data/cgsb/gresham/Niki/RESULTS/Growth/PRELIMINARY2/${TAG}.MAPQ15.bam OUTPUT=/data/cgsb/gresham/Niki/RESULTS/Growth/PRELIMINARY2/${TAG}.RNA_metrics ASSUME_SORTED=true REFERENCE_SEQUENCE=/data/cgsb/gresham/Niki/RESULTS/A111sequencing/References/ScERCC.fa CHART_OUTPUT=/data/cgsb/gresham/Niki/RESULTS/Growth/PRELIMINARY2/${TAG}.RNA_metrics.chart METRIC_ACCUMULATION_LEVEL=ALL_READS REF_FLAT=/data/cgsb/gresham/Niki/RESULTS/A111sequencing/References/sgdGene_refFlat STRAND_SPECIFICITY=SECOND_READ_TRANSCRIPTION_STRAND MINIMUM_LENGTH=150 

java -Xmx4g -jar /share/apps/picard-tools/1.111/CollectAlignmentSummaryMetrics.jar METRIC_ACCUMULATION_LEVEL=ALL_READS INPUT=/data/cgsb/gresham/Niki/RESULTS/Growth/PRELIMINARY2/${TAG}.MAPQ15.bam OUTPUT=/data/cgsb/gresham/Niki/RESULTS/Growth/PRELIMINARY2/${TAG}.ALN_Metrics ASSUME_SORTED=true REFERENCE_SEQUENCE=/data/cgsb/gresham/Niki/RESULTS/A111sequencing/References/ScERCC.fa
